azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/device_client.py

```diff
@@ -4,7 +4,30 @@
 # --------------------------------------------------------------------------------------------
 
 from .internal_client import InternalClient
+import logging
+from threading import Event
+
+logger = logging.getLogger(__name__)
 
 
 class DeviceClient(InternalClient):
-    pass
+    def __init__(self, auth_provider, transport):
+        InternalClient.__init__(self, auth_provider, transport)
+        self._transport.on_transport_c2d_message_received = self._handle_c2d_message_received
+        self.on_c2d_message = None
+
+    def enable_c2d_messages(self, handler_for_c2d=None):
```

Comment:
>enable_c2d_messages [](start = 8, length = 19)

since we're eventually going to have at least 4 "enable_X" functions and 4 "disable X" functions, I think it might make more sense to have an generic "enable_feature" function that takes a string that is the name of the feature.  Same for disable.

Reply:
can I be ignorantly nitpicky and suggest an enum rather than free-form string? :)

---
In reply to: [246541182](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246541182) [](ancestors = 246541182)

Reply:
i am absolutely in for enum. But the last time I did enum for the transport name there was a problem between version of python 2 and 3. 

---
In reply to: [246897708](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246897708) [](ancestors = 246897708,246541182)